tests/unit/test-cmd/test.build.js

```diff
@@ -64,301 +56,271 @@ describe('build', () => {
+      }, {fileFilter});
+      await zipFile.open(buildResult.extensionPath);
+      const fileNames = await zipFile.extractFilenames();
+      assert.notInclude(fileNames, 'background-script.js');
+      await zipFile.close();
+    }
+  ));
 
   it('lets you rebuild when files change', () => withTempDir(
-    (tmpDir) => {
+    async (tmpDir) => {
       const sourceDir = fixturePath('minimal-web-ext');
       const artifactsDir = tmpDir.path();
       const fileFilter = new FileFilter({sourceDir, artifactsDir});
       sinon.spy(fileFilter, 'wantFile');
       const onSourceChange = sinon.spy(() => {});
-      return build({
+      const buildResult = await build({
         sourceDir, artifactsDir, asNeeded: true,
       }, {
         manifestData: basicManifest, onSourceChange, fileFilter,
-      })
-        .then((buildResult) => {
-          // Make sure we still have a build result.
-          assert.match(buildResult.extensionPath, /\.zip$/);
-          return buildResult;
-        })
-        .then((buildResult) => {
-          assert.equal(onSourceChange.called, true);
-          const args = onSourceChange.firstCall.args[0];
-          assert.equal(args.sourceDir, sourceDir);
-          assert.equal(args.artifactsDir, artifactsDir);
-          assert.typeOf(args.onChange, 'function');
-
-          // Make sure it uses the file filter.
-          assert.typeOf(args.shouldWatchFile, 'function');
-          args.shouldWatchFile('/some/path');
-          assert.equal(fileFilter.wantFile.called, true);
-
-          // Remove the built extension.
-          return fs.unlink(buildResult.extensionPath)
-            // Execute the onChange handler to make sure it gets built
-            // again. This simulates what happens when the file watcher
-            // executes the callback.
-            .then(() => args.onChange());
-        })
-        .then((buildResult) => {
-          assert.match(buildResult.extensionPath, /\.zip$/);
-          return fs.stat(buildResult.extensionPath);
-        })
-        .then((stat) => {
-          assert.equal(stat.isFile(), true);
-        });
+      });
+      // Make sure we still have a build result.
+      assert.match(buildResult.extensionPath, /\.zip$/);
+      assert.equal(onSourceChange.called, true);
+      const args = onSourceChange.firstCall.args[0];
+      assert.equal(args.sourceDir, sourceDir);
+      assert.equal(args.artifactsDir, artifactsDir);
+      assert.typeOf(args.onChange, 'function');
+
+      // Make sure it uses the file filter.
+      assert.typeOf(args.shouldWatchFile, 'function');
+      args.shouldWatchFile('/some/path');
+      assert.equal(fileFilter.wantFile.called, true);
+
+      // Remove the built extension.
+      await fs.unlink(buildResult.extensionPath);
+      // Execute the onChange handler to make sure it gets built
+      // again. This simulates what happens when the file watcher
+      // executes the callback.
+      await args.onChange();
```

Comment:
This breaks the test Error: ENOENT: no such file or directory, stat '/var/folders/y_/qr18jygs6z56qqtmwnsn7nmh0000gn/T/tmp-web-ext-5923H6qh7ZsBc4oF/the_extension-0.0.1.zip'.
I think we need to await or rewrite this test